# The Vestibular System - Part 1: Mathematical Description

* 3 DOF Translations -> Otoliths (Utricle, Saccule) 
* 3 DOF Rotations -> 3 Semicircular Canals

<img src="data/images/labyrinth_labelled.png" style="width: 300px;"/>

## Semicircular Canals

The three semicircular canals (SCCs) on each side of the head transduce angular velocity. To describe this transduction mathematically, each canal can be described by a vector perpendicular to the surface spanned by
the canal $\vec{n}$.

### Orientation of all 3 Canals, on the Right Side of the Head


In [1]:
# Import the required packages
%matplotlib inline
import numpy as np
import matplotlib.pyplot as plt

# Make the display easy to read
% precision 2

'%.2f'

In [2]:
# Enter the information for the canalas into a Python dictionary.
# The data are taken from Della Santina et al,  "Orientation of Human Semicircular
# Canals Measured by Three-Dimensional Multi-planar CT Reconstruction.".
# J Assoc Res Otolaryngol 6(3): 191-206.

Canals = {'info': 'The matrix rows describe horizontal, anterior, and posterior canal orientation',
 'right': np.array(
        [[0.32269, -0.03837, -0.94573], 
         [0.58930,  0.78839,  0.17655],
         [0.69432, -0.66693,  0.27042]]),
 'left': np.array(
        [[-0.32269, -0.03837, 0.94573], 
         [-0.58930,  0.78839,  -0.17655],
         [-0.69432, -0.66693,  -0.27042]])}
        
# Normalize these vectors (just a small correction):
for side in ['right', 'left']:
    Canals[side] = (Canals[side].T / np.sqrt(np.sum(Canals[side]**2, axis=1))).T

# Show the results for the right SCCs:
print(Canals['info'])
print(Canals['right'])

The matrix rows describe horizontal, anterior, and posterior canal orientation
[[ 0.32 -0.04 -0.95]
 [ 0.59  0.79  0.18]
 [ 0.69 -0.67  0.27]]


### Example: Stimulation of the Right Horizontal Canal, by Rotation to the Right, with 100 deg/2

\begin{equation}
    stim = \vec{\omega} \cdot \vec{n}
\end{equation}

$\vec{\omega}$ ... relativ to the *head*!

<img src="data/images/SCC_stimulation_with_omega_.jpg" style="width: 150px;"/>



In [3]:
omega = np.r_[0, 0, -100]
stim = omega @ Canals['right'][0]

# Before Python 3.5:
# stim = np.dot(omega, Canals['right'][0])

print('The angular velocity sensed by the right horizontal canal is {0:3.1f} deg/s.'.format(stim))

The angular velocity sensed by the right horizontal canal is 94.6 deg/s.


<font color='red'>
<img src="data/images/matlab_logo_s.jpg" width=50px, align='left'>*Matlab Equivalent Commands*:
</font>

    % Define the semicircular canals
    Canals(1).side = 'right';
    Canals(1).rows = {'horizontal canal'; 'anterior canal'; 'posterior canal'};
    Canals(1).orientation = [	.365	 .158	-.905
        .652	 .753	-.017
        .757	-.561	 .320];
    Canals(2).side = 'left';
    Canals(2).rows = {'hor'; 'ant'; 'post'};
    Canals(2).orientation = [-.365	  .158 	 .905
        -.652	  .753 	 .017
        -.757 	 -.561	-.320];

    % Normalize the canal-vectors (only a tiny correction):
    for ii = 1:2
        for jj = 1:3
            Canals(ii).orientation(jj,:) = ...
                Canals(ii).orientation(jj,:) / norm( Canals(ii).orientation(jj,:) );
        end
    end

    % Calculate the stimulation
    omega = [0, 0, -100];
    stim = omega * Canals(1).orientation(1,:).T;

## Otoliths

The description of the otoliths is more involved, since each haircell on the otolith membrane senses the local strength of the gravito-inertal force. The direction of maximum sensitivity can for each haircell be described by a *Sensitivity Vector* $\vec{n}$.

<img src="data/images/UtriShapeHair.jpg" style="width: 600px;"/>


\begin{equation}
    \overrightarrow{GIF} = \vec{g} + \frac{d^2\vec{x}}{dt^2}
\end{equation}

\begin{equation}
    stim \approx \overrightarrow{GIF} \cdot \vec{n}
\end{equation}


- For the utricle, the sensitivity vectors lie approx. in the horizontal plane
- For the saccule, the sensitivity vectors lie approx. in the vertical plane


<img src="data/images/Forward.png" style="float: right;"/> [Vestibular_2_SCC_Transduction.ipynb](Vestibular_2_SCC_Transduction.ipynb)

    Author: Thomas Haslwanter
    Date: April 2018

<img src="data/images/CCA_40.png">
*The Vestibular System*  is licensed under a Creative Commons Attribution 4.0 International License.